CMakeLists.txt

```diff
@@ -1577,8 +1577,11 @@ endif()
 
 # TODO: - Fast huge math
 
+# Set processor-specific build macros
 if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "x86_64|AMD64")
     list(APPEND WOLFSSL_DEFINITIONS "-DWOLFSSL_X86_64_BUILD")
+elseif("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "aarch64|arm64")
```

Comment:
It looks like in configure.ac we just match host containing 'aarch64'. For WOLFSSL_X86_64_BUILD it does check though x86_64 and AMD64. Can we make this match the configure.ac checks?

Reply:
I think the `CMAKE_SYSTEM_PROCESSOR` variable is populated differently than configure's `$host_cpu` variable. On my M1 mac, it returns `arm64`, which fails this check. But in under autotools, the `aarch64` branch is hit in configure. I can investigate further to confirm

Reply:
ahhhh, yes if that is the case then disregard my comment

Reply:
@JacobBarthelmeh confirmed, the build system just represents them differently. [They appear to be the same](https://www.phoronix.com/news/MTY5ODk),  just a naming convention difference